llvm/lib/Target/AMDGPU/AMDGPUSubtarget.cpp

```diff
@@ -105,6 +105,14 @@ GCNSubtarget::initializeSubtargetDependencies(const Triple &TT,
                                         : AMDGPUSubtarget::SOUTHERN_ISLANDS;
   }
 
+  if (!hasFeature(AMDGPU::FeatureWavefrontSize32) &&
+      !hasFeature(AMDGPU::FeatureWavefrontSize64)) {
+    if (getGeneration() >= AMDGPUSubtarget::GFX10)
+      ToggleFeature(AMDGPU::FeatureWavefrontSize32);
+    else
+      ToggleFeature(AMDGPU::FeatureWavefrontSize64);
```

Comment:
Is it also true for GPUs like gfx8?

Reply:
Yes, all GCN before gfx10 were wave64 only.

Reply:
Oh okay. Good to know. I thought that is only a feature for MI series cards.